services/director-v2/src/simcore_service_director_v2/utils/dags.py

```diff
@@ -173,9 +173,13 @@ async def create_minimal_computational_graph_based_on_selection(
 async def compute_pipeline_details(
     complete_dag: nx.DiGraph, pipeline_dag: nx.DiGraph, comp_tasks: List[CompTaskAtDB]
 ) -> PipelineDetails:
-
-    # first pass, traversing in topological order to correctly get the dependencies, set the nodes states
-    await _set_computational_nodes_states(complete_dag)
+    try:
+        # FIXME: this problem of cyclic graphs for control loops create all kinds of issues that must be fixed
+        # first pass, traversing in topological order to correctly get the dependencies, set the nodes states
+        await _set_computational_nodes_states(complete_dag)
+    except nx.NetworkXUnfeasible:
```

Comment:
Found this snippet that could be handy to debug
```python
            problem_graph = ', '.join(f'{a} -> {b}'
                                      for a, b in nx.find_cycle(complete_dag))
            log.debug('Circular dependency detected: %s', problem_graph)
  ```

Reply:
networkx throws the exception when there is a cycle